### PR TITLE
Extending Draggable Context functionality

### PR DESCRIPTION
**Pull request by gedskikas**
# Description
Extending functionality to determine if a an item can be dropped on. This fixes an issue where line items could be dragged below the last existing line item, causing an issue shown in the screenshot below.

Before:
screen shot 2018-06-27 at 10 33 54

After:
screen shot 2018-06-27 at 15 16 55

# TODO
- [x] Release notes

# Testing Instructions

### CODE REVIEW

File: src/components/table/draggable-table-cell/draggable-table-cell.js [hunk 1]
```diff
@@ -19,12 +19,15 @@ const iconHTML = (
...
+  const canDrag = props.canDrag !== false;
+
   const icon = (
     <WithDrag
       identifier={ props.identifier }
       draggableNode={ props.draggableNode }
+      canDrag={ canDrag }
     >
-      { iconHTML }
+      { canDrag ? iconHTML : <span /> }
```
Comment (patrickmelvin) [inline src/components/table/draggable-table-cell/draggable-table-cell.js]: What's the purpose of this `span`?

Comment (gedskikas) [reply src/components/table/draggable-table-cell/draggable-table-cell.js]: To make sure the row isn't being rendered with an 'icon', but instead is just empty

**Comment by SeanArmstrong** [comment]
Added a issue around this area. It does not block the PR - https://github.com/Sage/carbon/issues/1786

**Comment by rpepe23** [comment]
Tested and it works as expected = OK 
Evidence in Jira SL-345

**Comment by gedskikas** [comment]
@SeanArmstrong @patrickmelvin @halezmo could you re-review please, had to make a small change.

**Pull request merged**